tests/src/text/test_format.cpp

```diff
@@ -0,0 +1,90 @@
+#include "es-test.hpp"
+
+// Include only the tested class's header.
+#include "../../../source/text/Format.h"
+
+// ... and any system includes needed for the test file.
+#include <string>
+
+namespace { // test namespace
+
+// #region mock data
+
+// Insert file-local data here, e.g. classes, structs, or fixtures that will be useful
+// to help test this class/method.
+
+// #endregion mock data
+
+
+
+// #region unit tests
+SCENARIO("A unit of playing time is to be made human-readable", "[Format][PlayTime]") {
+	GIVEN( "A time of 0" ) {
+		THEN( "0s is returned" ) {
+			CHECK( Format::PlayTime(0) == "0s");
+		}
+	}
+
+	GIVEN( "A time of a half second" ) {
+		THEN( "0s is returned" ) {
+			CHECK( Format::PlayTime(.5) == "0s");
+		}
+	}
+
+	GIVEN( "A time under a minute" ) {
+		THEN( "A time in only seconds is returned" ) {
+			CHECK( Format::PlayTime(47) == "47s");
+		}
+	}
+
+	GIVEN( "A time over a minute but under an hour " ) {
+		THEN( "A time in only minutes and seconds is returned" ) {
+			CHECK( Format::PlayTime(567) == "9m 27s");
+		}
+	}
+
+	GIVEN( "A time over an hour but under a day" ) {
+		THEN( "A time in only hours, minutes, and seconds is returned" ) {
+			CHECK( Format::PlayTime(8492) == "2h 21m 32s");
+		}
+	}
+
+	GIVEN( "A time over a day but under a year" ) {
+		THEN( "A time in only days, hours, minutes, and seconds is returned" ) {
+			CHECK( Format::PlayTime(5669274) == "65d 14h 47m 54s");
+		}
+	}
+
+	GIVEN( "A time over a year" ) {
+		THEN( "A time using all units is returned" ) {
+			CHECK( Format::PlayTime(98957582) == "3y 50d 8h 13m 2s");
+		}
+	}
+	GIVEN( "A negative time" ) {
+		THEN( "0s is returned " ) {
+			CHECK( Format::PlayTime(-300) == "0s");
+		}
+	}
+}
+
+// #endregion unit tests
+
+// #region benchmarks
+#ifndef CATCH_CONFIG_ENABLE_BENCHMARKING
+TEST_CASE( "Benchmark Format::PlayTime", "[!benchmark][random]" ) {
+	BENCHMARK( "Format::PlayTime() with a value under an hour" ) {
+		int randTime = Random::Int(3600)
```

Comment:
For more-comparable benchmarks (e.g. between different CI runs), we should remove explicit randomness